python/hsfs/feature_view.py

```diff
@@ -418,46 +415,583 @@ def create_training_dataset(
+    ):
+        if not (
+            (val_size and test_size)
+            or ((train_end or val_start) and (val_end or test_start))
+        ):
+            raise ValueError(
+                "Invalid split input."
+                "You should specify either (`val_size` and `test_size`)"
+                " or ((`train_end` or `val_start`) and (`val_end` or `test_start`))"
+            )
+
+    def get_training_dataset(
+        self,
+        training_dataset_version,
+        read_options: Optional[Dict[Any, Any]] = None,
+    ):
+        """
+        Get training data from storage or feature groups.
+
+        !!! info
+        If a materialised training data has deleted. Use `recreate_training_dataset()` to
+        recreate the training data.
+
+        # Arguments
+            version: training dataset version
+            read_options: Additional read options as key-value pairs, defaults to `{}`.
+                When using the `python` engine, read_options can contain the
+                following entries:
+                * key `spark` and value an object of type
+                [hsfs.core.job_configuration.JobConfiguration](../job_configuration)
+                  to configure the Hopsworks Job used to compute the training dataset.
+
+        # Returns
+            (X, y): Tuple of dataframe of features and labels
+
+        """
+        td, df = self._feature_view_engine.get_training_data(
+            self, read_options, training_dataset_version=training_dataset_version
+        )
+        return df
+
+    def get_train_test_splits(
+        self,
+        training_dataset_version,
+        read_options: Optional[Dict[Any, Any]] = None,
+    ):
+        """
+        Get training data from storage or feature groups.
+
+        !!! info
+        If a materialised training data has deleted. Use `recreate_training_dataset()` to
+        recreate the training data.
+
+        # Arguments
+            version: training dataset version
+            read_options: Additional read options as key-value pairs, defaults to `{}`.
+                When using the `python` engine, read_options can contain the
+                following entries:
+                * key `spark` and value an object of type
+                [hsfs.core.job_configuration.JobConfiguration](../job_configuration)
+                  to configure the Hopsworks Job used to compute the training dataset.
+
+        # Returns
+            (X_train, y_train, X_test, y_test):
+                Tuple of dataframe of features and labels
+
+        """
+        td, df = self._feature_view_engine.get_training_data(
+            self,
+            read_options,
+            training_dataset_version=training_dataset_version,
+            splits=[TrainingDatasetSplit.TRAIN, TrainingDatasetSplit.TEST],
+        )
+        return df[TrainingDatasetSplit.TRAIN] + df[TrainingDatasetSplit.TEST]
+
+    def get_train_validation_test_splits(
```

Comment:
I don't have a problem with splits here (instead of 'split'). 
It is more correct english.